### PR TITLE
Smooth Overpaint and Transparency

### PR DESCRIPTION
**Pull request by arose**
image

image

### CODE REVIEW

File: src/extensions/geo-export/mesh-exporter.ts [hunk 1]
```diff
@@ -184,11 +210,12 @@ export abstract class MeshExporter<D extends RenderObjectExportData> implements
         }
     }
 
-    protected static getColor(values: BaseValues, groups: Float32Array | Uint8Array, vertexCount: number, instanceIndex: number, isGeoTexture: boolean, interpolatedColors: Uint8Array | undefined, vertexIndex: number): Color {
+    protected static getColor(values: BaseValues, groups: Float32Array | Uint8Array, vertexCount: number, instanceIndex: number, isGeoTexture: boolean, interpolatedColors: Uint8Array | undefined, interpolatedOverpaint: Uint8Array | undefined, vertexIndex: number): Color {
```
Comment (dsehnal) [inline src/extensions/geo-export/mesh-exporter.ts:213]: Should we make it a convention in the code that a function that takes more than say 4 args should take an object parameter?

File: src/mol-geo/geometry/base.ts [hunk 2]
```diff
@@ -33,6 +34,40 @@ export const VisualQualityOptions = PD.arrayToOptions(VisualQualityNames);
...
+    }),
+};
+export type ColorSmoothingParams = typeof ColorSmoothingParams
+
+export function hasColorSmoothingProp(props: PD.Values<any>): props is PD.Values<ColorSmoothingParams> {
+    return !!props.smoothColors;
+}
+
+export function getColorSmoothingProps(smoothColors: PD.Values<ColorSmoothingParams>['smoothColors'], preferSmoothing?: boolean, resolution?: number) {
+    if ((smoothColors.name === 'on' || (smoothColors.name === 'auto' && preferSmoothing)) && resolution && resolution < 3) {
```
Comment (dsehnal) [inline src/mol-geo/geometry/base.ts:54]: What's the purpose of the ``resolution < 3`` check? Maybe make this a ``maxResolution`` option in ``auto`` and ``on``?

**Review by dsehnal — COMMENTED** [review]
Very nice!

File: src/extensions/geo-export/mesh-exporter.ts (hunk 1 above)
Comment (arose) [reply src/extensions/geo-export/mesh-exporter.ts:213]: :) I will consolidate those args

File: src/mol-geo/geometry/base.ts (hunk 2 above)
Comment (arose) [reply src/mol-geo/geometry/base.ts:54]: The smoothing uses vertex positions as sample points. At higher resolutions (of e.g. coarse gaussian surfaces) there are only a few vertices per color. In that case smoothing isn't helpful as the linear interpolation in the shader is enough. It is a bit of a magic number but until we see a problem with it I think we can keep it.

Comment (dsehnal) [reply src/mol-geo/geometry/base.ts:54]: Okay, makes sense.

**Pull request merged**